lib/logdna.rb

```diff
@@ -82,78 +59,69 @@ def level=(value)
       @level = value
     end
 
-    def log(msg=nil, opts={})
-      loggerExist?
-      message = msg
-      message = yield if msg.nil? && block_given?
-      @response = @@client.buffer(message, default_opts.merge(opts).merge({
-            timestamp: (Time.now.to_f * 1000).to_i
-        }))
-      'Saved'
+    def log(message = nil, opts = {})
+      if message.nil? && block_given?
+        message = yield
+      elsif message.nil?
+        puts "provide either a message or block"
+      end
+      message = message.to_s.encode("UTF-8")
+      @client.write_to_buffer(message, default_opts.merge(opts).merge(
+                                         timestamp: (Time.now.to_f * 1000).to_i
+                                       ))
     end
 
     Resources::LOG_LEVELS.each do |lvl|
       name = lvl.downcase
 
-      define_method name do |msg=nil, opts={}, &block|
-        self.log(msg, opts.merge({
-          level: lvl,
-        }), &block)
+      define_method name do |msg = nil, opts = {}, &block|
+        self.log(msg, opts.merge(
+                        level: lvl
+                      ), &block)
       end
 
       define_method "#{name}?" do
-        return Resources::LOG_LEVELS[self.level] == lvl if self.level.is_a? Numeric
+        return Resources::LOG_LEVELS[level] == lvl if level.is_a? Numeric
+
         self.level == lvl
       end
     end
 
     def clear
-      @app = 'default'
-      @level = 'INFO'
+      @app = "default"
+      @level = "INFO"
       @env = nil
       @meta = nil
     end
 
-    def loggerExist?
-      if @@client.nil?
-        puts "Logger Not Initialized Yet"
-        close
-      end
+    def <<(msg = nil, opts = {})
+      log(msg, opts.merge(
+                 level: ""
+               ))
     end
 
-    def <<(msg=nil, opts={})
-      self.log(msg, opts.merge({
-        level: '',
-      }))
-    end
-
-    def add(*arg)
+    def add(*_arg)
       puts "add not supported in LogDNA logger"
-      return false
+      false
     end
 
-    def unknown(msg=nil, opts={})
-      self.log(msg, opts.merge({
-        level: 'UNKNOWN',
-      }))
+    def unknown(msg = nil, opts = {})
+      log(msg, opts.merge(
+                 level: "UNKNOWN"
+               ))
     end
 
-    def datetime_format(*arg)
+    def datetime_format(*_arg)
       puts "datetime_format not supported in LogDNA logger"
-      return false
+      false
     end
 
-
     def close
-      if defined? @@client and !@@client.nil?
-          @@client.exitout()
-      end
+      @client.exitout if defined? @client && !@client.nil?
```

Comment:
Although optional, I prefer the style of always including parens on function calls for better readability.